File: mongodb/golang/routes/route.go
package routes

import (
	"note-golang-mongodb/controllers"
	"note-golang-mongodb/helpers"
	"note-golang-mongodb/repositories"
	"note-golang-mongodb/services"
	"note-golang-mongodb/utils"

	"github.com/labstack/echo/v4"
)

func SetRoute(e *echo.Echo, mongoUtil utils.MongoUtil, uuidHelper helpers.UuidHelper) {
	mongodbRepository := repositories.NewMongodbRepository()
	mongodbService := services.NewMongodbService(mongoUtil, uuidHelper, mongodbRepository)
	mongodbController := controllers.NewMongodbController(mongodbService)
	e.POST("/api/v1/test1", mongodbController.Create)
	e.GET("/api/v1/test1/:id", mongodbController.GetById)
	e.PUT("/api/v1/test1", mongodbController.UpdateById)
	e.DELETE("/api/v1/test1", mongodbController.DeleteOne)
}

File: mongodb/rust/src/routes/route.rs
use std::sync::Arc;

use axum::{routing::{post, get, put, delete}, Router};
use mongodb::Collection;

use crate::{controllers::test1_controller::{create, delete_by_id, get_by_id, update_by_id}, models::entities::test1::Test1, repositories::test1_repository::Test1RepositoryImpl, services::test1_service::Test1ServiceImpl, states::app_state::AppState, utils::mongo_util::{self, MongoUtilImpl}};

pub fn set_route(collection: Arc<Collection<Test1>>) -> Router {
    let test1_repository = Arc::new(Test1RepositoryImpl::new(collection));
    let test1_service = Arc::new(Test1ServiceImpl::new(test1_repository.clone()));
    let state = Arc::new(AppState{test1_service: test1_service.clone()});
    Router::new()
        .route("/api/v1/test1", post(create))
        .route("/api/v1/test1/{id}", get(get_by_id))
        .route("/api/v1/test1", put(update_by_id))
        .route("/api/v1/test1", delete(delete_by_id))
        .with_state(state)
}